argus/debug/debugger.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2014 Cloudbase Solutions Srl
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+import six
+import abc
+
+from multiprocessing.connection import Client
+
+@six.add_metaclass(abc.ABCMeta)
+class BaseDebugger(object):
+    """Abstract class for debugger.
+
+    Any debugger should at least have a pause method implemented
+    and actions before and after a function is executed.
+
+    Any debugger should inherit from this one.
+    """
+
+    def __init__(self, config):
+        """Initilaize and get needed config values to avoid importing util.
+        """
+        self._ip = config.ip
+        self._port = config.port
+        self._authkey = config.authkey
+
+    @abc.abstractmethod
+    def pause(self):
+        """The most basic action needed in a debugger it's pause.
+        """
+
```

Comment:
it's not needed. I took that from other argus code. https://github.com/cloudbase/cloudbase-init-ci/blob/master/argus/client/base.py#L36